src/main/java/org/folio/services/validator/InvoiceValidator.java

```diff
@@ -119,42 +126,76 @@ private void validateInvoiceLineFundDistributions(List<InvoiceLine> invoiceLines
       if (CollectionUtils.isEmpty(line.getFundDistributions())) {
         throw new HttpException(400, FUND_DISTRIBUTIONS_NOT_PRESENT);
       }
-
-      if (isFundDistributionSummaryNotValid(line, currencyUnit)) {
-        throw new HttpException(400, LINE_FUND_DISTRIBUTIONS_SUMMARY_MISMATCH);
-      }
+      validateFundDistributionForInvoiceLine(line.getSubTotal(),line.getFundDistributions(),currencyUnit);
     }
   }
 
-  private boolean isFundDistributionSummaryNotValid(InvoiceLine line, CurrencyUnit currencyUnit) {
-    return ObjectUtils.notEqual(sumMixedDistributions(line, currencyUnit), line.getTotal());
-  }
 
-  private Double sumMixedDistributions(InvoiceLine line, CurrencyUnit currencyUnit) {
-    return sumMixedDistributions(line.getFundDistributions(), line.getTotal(), currencyUnit);
-  }
-
-  private void validateInvoiceAdjustmentsDistributions(List<Adjustment> adjustments, CurrencyUnit currencyUnit) {
-    for (Adjustment adjustment : adjustments){
+  public static void validateInvoiceAdjustmentsDistributions(List<Adjustment> adjustments,CurrencyUnit currencyUnit) {
+    for (Adjustment adjustment : adjustments) {
       if (CollectionUtils.isEmpty(adjustment.getFundDistributions())) {
         throw new HttpException(400, ADJUSTMENT_FUND_DISTRIBUTIONS_NOT_PRESENT);
       }
-
-      if (isFundDistributionSummaryNotValid(adjustment, currencyUnit)) {
-        throw new HttpException(400, ADJUSTMENT_FUND_DISTRIBUTIONS_SUMMARY_MISMATCH);
+      validateFundDistributionForInvoiceLine(adjustment.getValue(), adjustment.getFundDistributions(), currencyUnit);
+    }
+  }
+  public static void validateFundDistributionForInvoiceLine(Double total, List<FundDistribution> fundDistributions,CurrencyUnit currencyUnit) {
```

Comment:
I am seeing a lot of `BigDecimal` conversion in this method and maybe this class in general, can we curb this somehow? Some operations appears to be possible without converting to a `BigDecimal` first.

Reply:
@okolawole-ebsco basically this logic is very similar to logic that we provided for order lines, and all this logic tested for orders, so I propose to leave as it for now.
Here is validation for orders: https://github.com/folio-org/mod-orders/blob/master/src/main/java/org/folio/orders/utils/FundDistributionUtils.java#L37

Reply:
I see. We can leave this for testing sake. My concern is about all the boxing and unboxing going on, which is performance related but I can't tell the impact from looking at the code.

An approach i would have taken is to have the POJO store as BigDecimal instead of Double since BigDecimal is what we want. I am certain, this change might break other things...